src/View/SSViewer_DataPresenter.php

```diff
@@ -303,6 +308,29 @@ public function getObj($name, $arguments = [], $cache = false, $cacheName = null
      */
     public function __call($name, $arguments)
     {
+        // $Count should be handled specially, so we can count raw arrays and iterables
+        // which aren't ViewableData.
+        if (empty($arguments) && ($name === 'Count' || $name === 'count')) {
+            $item = $this->getItem();
+            $result = null;
+            if ($item instanceof ViewableData) {
+                // Respect ViewableData casting
+                $result = $item->obj($name, [], true);
+            } elseif (is_object($item)) {
+                // Get the method or property from objects, if there is one
+                if (ClassInfo::hasMethod($item, $name)) {
+                    $result = $item->$name();
+                } elseif (isset($item->$name)) {
+                    $result = $item->$name;
+                }
+            } elseif (is_countable($item)) {
+                // Count countables
+                $result = count($item);
+            }
+            $this->resetLocalScope();
+            return $result;
+        }
```

Comment:
This allows the use of `$Count` in templates to work the same for arrays and other iterators as it would for any `SS_List` implementation.

It still respects type casting in `ViewableData` when the current item in scope (the item this is being called on) is an instance of `ViewableData`.